src/views/virtualmachines/list/components/VirtualMachineRow/VirtualMachineRowLayout.tsx

```diff
@@ -49,7 +51,19 @@ const VirtualMachineRowLayout: FC<
   const vmName = useMemo(() => getName(obj), [obj]);
   const vmNamespace = useMemo(() => getNamespace(obj), [obj]);
 
-  const storageClasses = useMemo(() => getVirtualMachineStorageClasses(obj), [obj]);
```

Comment:
1. it seems we still need a helper function as this code pattern is repeated in other places
2. `reduce()`  wraps some procedural-style logic - that's OK but I would keep it one style i.e. 
```
getVolumes(vm)
  .map(volume => volume.dataVolume?.name || volume.persistentVolumeClaim?.claimName)
  .map(name => pvcMapper?.[vmNamespace]?.[name])
  .map(volumePVC => volumePVC?.spec?.storageClassName)
  .filter(Boolean)
// + some deduplication
```